tap_hubspot/__init__.py

```diff
@@ -252,17 +265,13 @@ def parse_source_from_url(url):
         return match.group(1)
     return None
 
-
-@backoff.on_exception(backoff.constant,
-                      (requests.exceptions.RequestException,
-                       requests.exceptions.HTTPError),
-                      max_tries=5,
-                      jitter=None,
-                      giveup=giveup,
-                      on_giveup=on_giveup,
-                      interval=10)
-def request(url, params=None):
-
+def get_params_and_headers(params):
+    """
+    This function makes a params object and headers object based on the
+    authentication values available. If there is an `hapikey` in the config, we
+    need that in `params` and not in the `headers`. Otherwise, we need to get an
+    `access_token` to put in the `headers` and not in the `params`
+    """
     params = params or {}
```

Comment:
feels like this could also just be a default value, but unfortunately dict default values can be modified which linters catch... so just leave it like this...